src/v/archival/scrubber.cc

```diff
@@ -30,8 +31,20 @@ scrubber::scrubber(
   , _feature_table(feature_table)
   , _detector{_archiver.get_bucket_name(), _archiver.get_ntp(), _archiver.get_revision_id(), _remote, _logger, _as}
   , _scheduler(
-      [this] { return _archiver.manifest().last_partition_scrub(); },
-      std::move(interval),
+      [this] {
+          const auto at = _archiver.manifest().last_partition_scrub();
+          const auto offset = _archiver.manifest().last_scrubbed_offset();
+          cloud_storage::scrub_status status;
+          if (!offset && at != model::timestamp::missing()) {
+              status = cloud_storage::scrub_status::full;
+          } else {
+              status = cloud_storage::scrub_status::partial;
```

Comment:
Sure. I'll add a comment there.

Note that this code does not determine whether we should perform a full or a partial scrub. Instead, it tells the scheduler if the last scrub was full (reached the end of the log) or partial (stopped somewhere before the end of the log).

We know that we've done a "full scrub" if the last scrubbed offset is set to null and the timestamp has a value.
Both are needed because the default values are null for the last offset and `missing` for the timestamp.

Reply:
Added a comment.